### PR TITLE
Add a doc on how to include an upstream Theia patch into the Che-Theia release

### PR DESCRIPTION
**Pull request by azatsarynnyy**
Signed-off-by: Artem Zatsarynnyi <azatsary@redhat.com>

### What does this PR do?
Adds a doc on how to include an upstream Theia patch into the Che-Theia release.

### What issues does this PR fix or reference?

closes https://github.com/eclipse/che/issues/19141, https://issues.redhat.com/browse/RHDEVDOCS-2741

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -1,3 +1,29 @@
-# Eclipse Che-Theia release process
+## Eclipse Che-Theia release process
+See [Release Che-Theia](./.github/workflows/release.yml) workflow.
 
-See [Release Che Theia](./.github/workflows/release.yml) workflow.
+## Bugfix release with the upstream Theia commit included
+To make another bugfix release of Che-Theia with the upstream Theia patch included, there're several options available.
+
+### To include all the commits made before the required patch
+- update the `build.include` file for an interested Che-Theia tag:
```
Comment (benoitf) [inline RELEASE.md:8]:
```suggestion
- update the `build.include` file to match upstream Eclipse Theia revision:
```

File: RELEASE.md [hunk 2]
```diff
@@ -1,3 +1,29 @@
...
+  - `THEIA_COMMIT_SHA="<commit-sha>"`
+- continue the release process as usual
+
+### To include only the required upstream commit
+- clone the Theia fork https://github.com/redhat-developer/codeready-workspaces-theia
```
Comment (benoitf) [inline RELEASE.md:13]: what fork is it ? theia or che-theia or codeready workspaces ?

File: RELEASE.md [hunk 3]
```diff
@@ -1,3 +1,29 @@
...
+- sync the `master` branch of the forked repository with the origin one and push the changes
+- take the Theia commit sha which was used for an interested Che-Theia release, e.g. for 7.27.x it's [`1110f990`](https://github.com/eclipse/che-theia/blob/7.27.x/build.include#L17)
+- create a branch
+- cherry-pick the required commit sha to the created branch and push the changes
+- update the `build.include` file for an interested Che-Theia tag:
```
Comment (benoitf) [inline RELEASE.md:18]: `an interested Che-Theia tag` looks odd

**Review by benoitf — COMMENTED** [review]
nice doc.

I think you should run the content against che-docs vale plug-in to remove all past tense form

The Theia or Che-Theia fork is not clear to me (there is codeready workspaces example)

**Comment by dmytro-ndp** [comment]
[crw-ci-test --rebuild]

**Comment by azatsarynnyy** [comment]
> I think you should run the content against che-docs vale plug-in to remove all past tense form

@benoitf the only way I know how to check it with the che-docs vale plug-in - it's open a draft PR and look at the check's output.
Do you know a simpler or quicker way?

File: RELEASE.md (hunk 2 above)
Comment (azatsarynnyy) [reply RELEASE.md]: It was the wrong link. I updated it with the correct one.

File: RELEASE.md (hunk 3 above)
Comment (azatsarynnyy) [reply RELEASE.md]: agree :slightly_smiling_face: 
rewritten

File: RELEASE.md [hunk 4]
```diff
@@ -1,3 +1,30 @@
...
+- continue the release process as usual
+
+### To include only the required upstream commit
+- clone the Theia fork https://github.com/redhat-developer/eclipse-theia
+- sync the `master` branch of the forked repository with the origin one and push the changes
+- take the Theia commit sha which was used for a Che-Theia release, e.g. for 7.27.x it's [`1110f990`](https://github.com/eclipse/che-theia/blob/7.27.x/build.include#L17)
+- create a new branch
+- cherry-pick the required commit sha to the created branch and push the changes
+- update the `build.include` file:
+  - `THEIA_GITHUB_REPO="redhat-developer/codeready-workspaces-theia"`
```
Comment (benoitf) [inline RELEASE.md:20]: I think here it should be `redhat-developer/eclipse-theia` no ?

Comment (benoitf) [inline RELEASE.md:20]:
```suggestion
  - `THEIA_GITHUB_REPO="redhat-developer/eclipse-theia"`
```

**Review by benoitf — APPROVED** [review]

File: RELEASE.md (hunk 4 above)
Comment (azatsarynnyy) [reply RELEASE.md]: sure, thx!

**Pull request merged**